framework/DataObjects/DataSet.py

```diff
@@ -1627,20 +1628,18 @@ def _setScalingFactors(self,var=None):
         pass
     # TODO someday make KDTree too!
     assert(self._data is not None) # TODO check against collector entries?
-    for var in varList:
-      ## commented code. We use a try now for speed. It probably needs to be modified for ND arrays
-      # if not a float or int, don't scale it
-      # TODO this check is pretty convoluted; there's probably a better way to figure out the type of the variable
-      #first = self._data.groupby(var).first()[var].item(0)
-      #if (not mathUtils.isAFloatOrInt(first)) or np.isnan(first):# or self._data[var].isnull().all():
-      #  continue
-      try:
-        mean = float(self._data[var].mean())
-        scale = float(self._data[var].std())
-        self._scaleFactors[var] = (mean,scale)
-      except Exception:
-        self.raiseADebug('Had an issue with setting scaling factors for variable "{}". No big deal.'.format(var))
-        pass
+    ds = self._data[varList] if var is not None else self._data
+    mean = ds.mean().variables
+    scale = ds.std().variables
+    for name, val in mean.items():
+      if name in varList:
+        try:
+          m = float(val.values)
+          s = float(scale[name].values)
```

Comment:
are the "float" casts needed here if "mean()" and "std()" are called above?

Reply:
probably ok, the returns are the 0-dimensional numpy array without casting.

Reply:
Oh, I see. It might be faster to take the zeroth element instead of float casting, but I'm not 100% sure of that.

Reply:
First, in order to get the values, we can use ```[()]```. Second, both cases should works for histories and ND entries, unless we want to scalar our data along given dimensions.